quickdump: warning on arm32

```sh
/mnt/data/libelektra/src/plugins/quickdump/quickdump.c: In function ‘elektraQuickdumpGet’:
/mnt/data/libelektra/src/plugins/quickdump/quickdump.c:178:28: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  178 |  while ((c = fgetc (file)) != EOF)
      |                            ^~
/mnt/data/libelektra/src/plugins/quickdump/quickdump.c:192:12: warning: comparison is always false due to limited range of data type [-Wtype-limits]
  192 |   if (type == EOF)
      |            ^~
/mnt/data/libelektra/src/plugins/quickdump/quickdump.c:263:10: warning: comparison is always false due to limited range of data type [-Wtype-limits]
  263 |    if (c == EOF)
      |          ^~
```

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -73,6 +73,10 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 
 - Minor improvement of source code readability in uname.c _(@lawli3t)_
 
+### quickdump
+
+- Fixed an issue with type-limits on ARM32 (see issue #4217). _(Klemens Böswirth @kodebach)_
+
 ### <<Plugin6>>
 
 - <<TODO>>

--- a/src/plugins/quickdump/quickdump.c
+++ b/src/plugins/quickdump/quickdump.c
@@ -174,9 +174,10 @@ int elektraQuickdumpGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key 
 	nameBuffer.string[parentSize] = '\0';	 // set new null terminator
 	nameBuffer.offset = parentSize;		 // set offset to null terminator
 
-	char c;
-	while ((c = fgetc (file)) != EOF)
+	int fc;
+	while ((fc = fgetc (file)) != EOF)
 	{
+		char c = fc;
 		ungetc (c, file);
 
 		if (!readStringIntoBuffer (file, &nameBuffer, parentKey))
@@ -188,8 +189,8 @@ int elektraQuickdumpGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key 
 			return ELEKTRA_PLUGIN_STATUS_ERROR;
 		}
 
-		char type = fgetc (file);
-		if (type == EOF)
+		int ftype = fgetc (file);
+		if (ftype == EOF)
 		{
 			elektraFree (nameBuffer.string);
 			elektraFree (metaNameBuffer.string);
@@ -199,6 +200,7 @@ int elektraQuickdumpGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key 
 			return ELEKTRA_PLUGIN_STATUS_ERROR;
 		}
 
+		char type = ftype;
 		Key * k;
 
 		switch (type)
@@ -258,15 +260,16 @@ int elektraQuickdumpGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key 
 			return ELEKTRA_PLUGIN_STATUS_ERROR;
 		}
 
-		while ((c = fgetc (file)) != 0)
+		while ((fc = fgetc (file)) != 0)
 		{
-			if (c == EOF)
+			if (fc == EOF)
 			{
 				keyDel (k);
 				fclose (file);
 				ELEKTRA_SET_VALIDATION_SYNTACTIC_ERROR (parentKey, "Missing key end");
 				return ELEKTRA_PLUGIN_STATUS_ERROR;
 			}
+			c = fc;
 
 			switch (c)
 			{

